electron/main/backend.ts

```diff
@@ -0,0 +1,85 @@
+import * as childProcess from "child_process";
+import { join, resolve } from "path";
+import treeKill from "tree-kill";
+
+const sendBackendLogToStudio = (win: Electron.BrowserWindow, data: string) => {
+  if (global.initializingBackend) {
+    win.webContents.send("backend", data);
+  }
+};
+
+export const runCommand = (
+  command: string,
+  matchText: string,
+  win: Electron.BrowserWindow,
+): Promise<{
+  script: childProcess.ChildProcess;
+}> => {
+  return new Promise((resolve, reject) => {
+    const script = childProcess.exec(command);
+    script.stdout?.on("data", function (data) {
+      sendBackendLogToStudio(win, data);
+      if (data.toString().includes(matchText)) {
+        win.webContents.send("backend", "backend initialized successfully!");
+        resolve({ script });
+      }
+    });
+    script.stderr?.on("data", function (data) {
+      sendBackendLogToStudio(win, data);
+      if (data.toString().includes(matchText)) {
+        sendBackendLogToStudio(win, "backend initialized successfully!");
+        resolve({ script });
+      }
+    });
```

Comment:
The function name makes it seem like it's supposed to be a general function for running any command, but the logging from `win.webContents.send` show that it's only supposed to be used for backend initialization?